several compile issues on Fedora 30

## Problem description

Compilation errors as follows:

1)
rc/match/eis-bwtseq-context.c: In function \u2018gt_BWTSCRFGet\u2019:
src/match/eis-bwtseq-context.c:291:39: error: \u2018cxm\u2019 directive output may be truncated writing 3 bytes into a region of size between 2 and 6 [-Werror=format-truncation=]
  291 |       snprintf(buf, sizeof (buf), ".%ucxm", (unsigned)mapIntervalLog2);
      |                                       ^~~
src/match/eis-bwtseq-context.c:291:7: note: \u2018snprintf\u2019 output between 6 and 10 bytes into a destination of size 8
  291 |       snprintf(buf, sizeof (buf), ".%ucxm", (unsigned)mapIntervalLog2);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/match/eis-bwtseq-context.c: In function \u2018gt_BWTSeqCRLoad\u2019:
src/match/eis-bwtseq-context.c:291:39: error: \u2018cxm\u2019 directive output may be truncated writing 3 bytes into a region of size between 2 and 6 [-Werror=format-truncation=]
  291 |       snprintf(buf, sizeof (buf), ".%ucxm", (unsigned)mapIntervalLog2);
      |                                       ^~~
src/match/eis-bwtseq-context.c:291:7: note: \u2018snprintf\u2019 output between 6 and 10 bytes into a destination of size 8
  291 |       snprintf(buf, sizeof (buf), ".%ucxm", (unsigned)mapIntervalLog2);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/match/eis-bwtseq-context.c:291:39: error: \u2018cxm\u2019 directive output may be truncated writing 3 bytes into a region of size between 2 and 6 [-Werror=format-truncation=]
  291 |       snprintf(buf, sizeof (buf), ".%ucxm", (unsigned)mapIntervalLog2);
      |                                       ^~~
src/match/eis-bwtseq-context.c:291:7: note: \u2018snprintf\u2019 output between 6 and 10 bytes into a destination of size 8
  291 |       snprintf(buf, sizeof (buf), ".%ucxm", (unsigned)mapIntervalLog2);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:811: obj/src/match/eis-bwtseq-context.o] Error 1


2)
src/ltr/ltr_cluster_prepare_seq_visitor.c: In function \u2018gt_ltr_cluster_prepare_seq_visitor_feature_node\u2019:
src/ltr/ltr_cluster_prepare_seq_visitor.c:146:39: error: \u2018%lu\u2019 directive output may be truncated writing between 1 and 20 bytes into a region of size between 0 and 8191 [-Werror=format-truncation=]
  146 |       (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer, range.start,
      |                                       ^~~~~
In file included from /usr/local/src/GENOMETOOLS/OLD/genometools-1.5.9/src/core/array_api.h:23,
                 from /usr/local/src/GENOMETOOLS/OLD/genometools-1.5.9/src/core/array.h:21,
                 from src/ltr/ltr_cluster_prepare_seq_visitor.c:19:
/usr/local/src/GENOMETOOLS/OLD/genometools-1.5.9/src/core/types_api.h:55:16: note: format string is defined here
   55 | #define GT_WU "%lu"
      |                ^~~
src/ltr/ltr_cluster_prepare_seq_visitor.c:146:14: note: \u2018snprintf\u2019 output between 5 and 8234 bytes into a destination of size 8192
  146 |       (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer, range.start,
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  147 |                       range.end);
      |                       ~~~~~~~~~~
src/ltr/ltr_cluster_prepare_seq_visitor.c:109:39: error: \u2018%lu\u2019 directive output may be truncated writing between 1 and 20 bytes into a region of size between 0 and 8191 [-Werror=format-truncation=]
  109 |       (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer, range.start,
      |                                       ^~~~~
In file included from /usr/local/src/GENOMETOOLS/OLD/genometools-1.5.9/src/core/array_api.h:23,
                 from /usr/local/src/GENOMETOOLS/OLD/genometools-1.5.9/src/core/array.h:21,
                 from src/ltr/ltr_cluster_prepare_seq_visitor.c:19:
/usr/local/src/GENOMETOOLS/OLD/genometools-1.5.9/src/core/types_api.h:55:16: note: format string is defined here
   55 | #define GT_WU "%lu"
      |                ^~~
src/ltr/ltr_cluster_prepare_seq_visitor.c:109:14: note: \u2018snprintf\u2019 output between 5 and 8234 bytes into a destination of size 8192
  109 |       (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer, range.start,
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  110 |                       range.end);
      |                       ~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:811: obj/src/ltr/ltr_cluster_prepare_seq_visitor.o] Error 1


3)
src/ltr/ltr_cluster_stream.c: In function \u2018process_feature\u2019:
src/ltr/ltr_cluster_stream.c:352:41: error: \u2018%lu\u2019 directive output may be truncated writing between 1 and 20 bytes into a region of size between 0 and 8191 [-Werror=format-truncation=]
  352 |         (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer,
      |                                         ^~~~~
In file included from /usr/local/src/GENOMETOOLS/OLD/genometools-1.5.9/src/core/array_api.h:23,
                 from /usr/local/src/GENOMETOOLS/OLD/genometools-1.5.9/src/core/array.h:21,
                 from src/ltr/ltr_cluster_stream.c:19:
/usr/local/src/GENOMETOOLS/OLD/genometools-1.5.9/src/core/types_api.h:55:16: note: format string is defined here
   55 | #define GT_WU "%lu"
      |                ^~~
src/ltr/ltr_cluster_stream.c:352:16: note: \u2018snprintf\u2019 output between 5 and 8234 bytes into a destination of size 8192
  352 |         (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  353 |                         range.start, range.end);
      |                         ~~~~~~~~~~~~~~~~~~~~~~~
src/ltr/ltr_cluster_stream.c:344:41: error: \u2018%lu\u2019 directive output may be truncated writing between 1 and 20 bytes into a region of size between 0 and 8191 [-Werror=format-truncation=]
  344 |         (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer,
      |                                         ^~~~~
In file included from /usr/local/src/GENOMETOOLS/OLD/genometools-1.5.9/src/core/array_api.h:23,
                 from /usr/local/src/GENOMETOOLS/OLD/genometools-1.5.9/src/core/array.h:21,
                 from src/ltr/ltr_cluster_stream.c:19:
/usr/local/src/GENOMETOOLS/OLD/genometools-1.5.9/src/core/types_api.h:55:16: note: format string is defined here
   55 | #define GT_WU "%lu"
      |                ^~~
src/ltr/ltr_cluster_stream.c:344:16: note: \u2018snprintf\u2019 output between 5 and 8234 bytes into a destination of size 8192
  344 |         (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  345 |                         range.start, range.end);
      |                         ~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:811: obj/src/ltr/ltr_cluster_stream.o] Error 1


NOTE: all issues are easily fixed by increasing the buffers, but I have not looked into what the correct minimum size should be.
## Did you compile GenomeTools from source? If so, please state the `make` parameters used.
make using the following compiler
   gcc (GCC) 9.1.1 20190503 (Red Hat 9.1.1-1)


## What operating system (e.g. Ubuntu, Mac OS X), OS version (e.g. 15.10, 10.11) and platform (e.g. x86_64) are you using?
Fedora 30, x86_64

--- a/.travis.yml
+++ b/.travis.yml
@@ -42,13 +42,13 @@ matrix:
       env: opt=no cairo=no
     - os: osx
       compiler: gcc
-      env: default=yes
+      env: default=yes PKG_CONFIG_PATH="/usr/local/opt/libffi/lib/pkgconfig"
     - os: osx
       compiler: gcc
-      env: assert=no
+      env: assert=no PKG_CONFIG_PATH="/usr/local/opt/libffi/lib/pkgconfig"
     - os: osx
       compiler: gcc
-      env: amalgamation=yes
+      env: amalgamation=yes PKG_CONFIG_PATH="/usr/local/opt/libffi/lib/pkgconfig"
     - os: linux
       compiler: gcc
       env: opt=no cairo=no

--- a/scripts/travis_installdeps.sh
+++ b/scripts/travis_installdeps.sh
@@ -9,7 +9,8 @@ if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then
   sudo apt-get install binutils-mingw-w64-x86-64 gcc-mingw-w64-x86-64 -y
 else
   brew update
-  brew install pango cairo 
+  brew install pkg-config
+  brew install pango cairo libffi
   brew unlink proj
   brew install blast --force
 fi

--- a/src/ltr/ltr_cluster_prepare_seq_visitor.c
+++ b/src/ltr/ltr_cluster_prepare_seq_visitor.c
@@ -101,12 +101,12 @@ static int gt_ltr_cluster_prepare_seq_visitor_feature_node(GtNodeVisitor *nv,
       GtRange range;
       const char *attr;
       GtEncseqBuilder *eb;
-      char header[BUFSIZ];
+      char header[2*BUFSIZ];
       attr = gt_feature_node_get_attribute(curnode, "name");
       if (!attr)
         continue;
       range = gt_genome_node_get_range((GtGenomeNode*) curnode);
-      (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer, range.start,
+      (void) snprintf(header, 2*BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer, range.start,
                       range.end);
       if (!gt_hashmap_get(lcv->encseq_builders, attr)) {
         eb = gt_encseq_builder_new(gt_encseq_alphabet(lcv->src_encseq));
@@ -129,7 +129,7 @@ static int gt_ltr_cluster_prepare_seq_visitor_feature_node(GtNodeVisitor *nv,
       char *tmp;
       GtRange range;
       GtEncseqBuilder *eb;
-      char header[BUFSIZ];  /* XXX: use GtStr for safety */
+      char header[2*BUFSIZ];  /* XXX: use GtStr for safety */
       if (strcmp(fnt, gt_ft_long_terminal_repeat) == 0) {
         if (first_ltr) {
           tmp = gt_cstr_dup("lLTR");
@@ -143,7 +143,7 @@ static int gt_ltr_cluster_prepare_seq_visitor_feature_node(GtNodeVisitor *nv,
         gt_free(tmp);
         continue;
       }
-      (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer, range.start,
+      (void) snprintf(header, 2*BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer, range.start,
                       range.end);
       if (!gt_hashmap_get(lcv->encseq_builders, tmp)) {
         eb = gt_encseq_builder_new(gt_encseq_alphabet(lcv->src_encseq));

--- a/src/ltr/ltr_cluster_stream.c
+++ b/src/ltr/ltr_cluster_stream.c
@@ -321,7 +321,7 @@ static int cluster_annotate_nodes(GtClusteredSet *cs, GtEncseq *encseq,
       continue;
     fni = gt_feature_node_iterator_new((GtFeatureNode*) gn);
     while ((curnode = gt_feature_node_iterator_next(fni)) != NULL) {
-      char header[BUFSIZ];
+      char header[2*BUFSIZ];
       fnt = gt_feature_node_get_type(curnode);
       if (strcmp(fnt, gt_ft_repeat_region) == 0) {
         const char *rid;
@@ -341,7 +341,7 @@ static int cluster_annotate_nodes(GtClusteredSet *cs, GtEncseq *encseq,
         range = gt_genome_node_get_range((GtGenomeNode*) curnode);
         if ((range.end - range.start + 1) < 10UL)
           continue;
-        (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer,
+        (void) snprintf(header, 2*BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer,
                         range.start, range.end);
         gt_hashmap_add(desc2node, (void*) gt_cstr_dup(header), (void*) curnode);
       } else if (strcmp(fnt, real_feature) == 0) {
@@ -349,7 +349,7 @@ static int cluster_annotate_nodes(GtClusteredSet *cs, GtEncseq *encseq,
         range = gt_genome_node_get_range((GtGenomeNode*) curnode);
         if ((range.end - range.start + 1) < 10UL)
           continue;
-        (void) snprintf(header, BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer,
+        (void) snprintf(header, 2*BUFSIZ, "%s_"GT_WU"_"GT_WU"", buffer,
                         range.start, range.end);
         gt_hashmap_add(desc2node, (void*) gt_cstr_dup(header), (void*) curnode);
       }

--- a/src/match/eis-bwtseq-context.c
+++ b/src/match/eis-bwtseq-context.c
@@ -287,7 +287,7 @@ BWTSeqCRMapOpen(unsigned short mapIntervalLog2,
                             seqLen, mapIntervalLog2));
     mapName = gt_str_new_cstr(projectName);
     {
-      char buf[1 + 4 + 3 + 2];
+      char buf[1 + 4 + 3 + 3];
       snprintf(buf, sizeof (buf), ".%ucxm", (unsigned)mapIntervalLog2);
       gt_str_append_cstr(mapName, buf);
       if (createMapFile)

